frontend/javascripts/viewer/model/sagas/saving/save_mutex_saga.tsx

```diff
@@ -337,20 +353,26 @@ function* watchForActiveToolChange(mutexLogicState: MutexLogicState): Saga<void>
       const newActiveTool = yield* select((state) => state.uiInformation.activeTool);
       newToolId = newActiveTool.id;
     }
-    const newOnlyRequiredOnSave = TOOLS_WITH_ON_DEMAND_MUTEX_SUPPORT.includes(newToolId);
-    if (newOnlyRequiredOnSave !== mutexLogicState.onlyRequiredOnSave) {
-      mutexLogicState.onlyRequiredOnSave = newOnlyRequiredOnSave;
+
+    // TODO: also check for livecollab flag and editable mapping locked and so on.
```

Comment:
todo

Reply:
This is all now done in `restartMutexAcquiringSaga` -> `startSagaWithAppropriateMutexFetchingStrategy` -> `getCurrentMutexFetchingStrategy`. This makes the code behave consistent compared to the other actions which might restart the mutex fetching.